libstuff/SFastBuffer.cpp

```diff
@@ -8,6 +7,58 @@ SFastBuffer::SFastBuffer() : front(0) {
 SFastBuffer::SFastBuffer(const string& str) : front(0), data(str) {
 }
 
+bool SFastBuffer::startsWithHTTPRequest() {
+    // No HTTP request is less than 4 bytes. Strictly, an HTTP request is longer than this, but this is all we need to care about.
+    if (size() < 4) {
+        return false;
+    }
+
+    // Do we have headers yet? If not, keep looking for them.
+    // Headers are optional, but this will actually contain the methodline as well, so we won't end up with an ambiguous case where '0' means both "we haven't found them yet" and "there
+    // aren't any".
+    if (!headerLength) {
+        size_t next = nextToCheck;
+        while (next != string::npos && !headerLength) {
```

Comment:
Do we need the `next != string::npos` here? We are breaking below on that condition already